yarn-project/simulator/src/client/client_execution_context.ts

```diff
@@ -133,31 +138,49 @@ export class ClientExecutionContext extends ViewDataOracle {
   }
 
   /**
-   * Return the encrypted logs emitted during this execution.
+   * Return the note encrypted logs emitted during this execution.
    */
-  public getEncryptedLogs() {
-    return this.encryptedLogs;
+  public getNoteEncryptedLogs() {
+    // Do not return logs that have been chopped in the cache
+    const filteredLogs = this.noteEncryptedLogs.filter(l => this.noteCache.getLogs().includes(l));
+    return filteredLogs;
+  }
+
+  /**
+   * Sometimes notes can be chopped after a nested execution is complete.
+   * This means finished nested executions still hold transient logs. This method removes them.
+   * TODO(Miranda): is there a cleaner solution?
+   */
+  public chopNoteEncryptedLogsFromNested() {
+    // Do not return logs that have been chopped in the cache
+    const allNoteLogs = this.noteCache.getLogs();
+    this.nestedExecutions.forEach(result => {
+      if (!result.noteEncryptedLogs[0]?.isEmpty()) {
+        // The execution has note logs
+        result.noteEncryptedLogs = result.noteEncryptedLogs.filter(l => allNoteLogs.includes(l));
+      }
+    });
```

Comment:
I'm unsure about this solution - we do need some way of ensuring a tx does not contain logs chopped in the same call. E.g. a call can have two nested calls, the first adding a note and note log, the second nullifying it. The parent call will still gather the chopped note log from `.nestedExecutions` because that exists as a complete `ExecutionResult` which was finalised before the nullifier existed.
I added this here because:

- In `pxe_service`, where logs are added to a tx, we only have access to `ExecutionResult`s and in most cases final log hashes, not individual logs from the tail circuit.
- In `ExecutionResult`, where logs are collected, we could look through all nested executions for nullifiers and link them to logs, but this would be a much longer and messier solution (and means returns from `private_execution` would be incorrect until we gather logs later on).